llpc/context/llpcCompiler.cpp

```diff
@@ -1717,11 +1781,24 @@ Result Compiler::BuildComputePipeline(
     constexpr uint32_t ShaderCacheCount = 2;
     ShaderCache* pShaderCache[ShaderCacheCount] = { nullptr, nullptr };
     CacheEntryHandle hEntry[ShaderCacheCount] = { nullptr, nullptr };
-
-    cacheEntryState = LookUpShaderCaches(pPipelineInfo->pShaderCache, &cacheHash, &elfBin, pShaderCache, hEntry);
+    if (!buildingRelocatableElf)
+    {
+        cacheEntryState = LookUpShaderCaches(pPipelineInfo->pShaderCache, &cacheHash, &elfBin, pShaderCache, hEntry);
+    }
+    else
+    {
+        cacheEntryState = ShaderEntryState::Compiling;
+    }
 #else
     CacheEntryHandle hEntry = nullptr;
-    cacheEntryState = LookUpShaderCache(&cacheHash, &elfBin, &hEntry);
+    if (!buildingRelocatableElf)
+    {
+        cacheEntryState = LookUpShaderCache(&cacheHash, &elfBin, &hEntry);
+    }
+    else
+    {
+        cacheEntryState = ShaderEntryState::Compiling;
+    }
```

Comment:
Could you please first make a change that eliminates the preprocessor #if?

Reply:
This is a change that belong to #440 and #438. The preprocessor is pre-existing code related to the different handling of shader caches controlled by LLPC_CLIENT_INTERFACE_MAJOR_VERSION. Could you please advise how you want them eliminated?

Reply:
I ended up doing this myself. See #466